packages/core/scripts/collateral-umip/1_Propose.js

```diff
@@ -2,16 +2,37 @@
 // fork of the main net or can be run directly on the main net to execute the upgrade transactions.
 // To run this on the localhost first fork main net into Ganache with the proposerWallet unlocked as follows:
 // ganache-cli --fork https://mainnet.infura.io/v3/5f56f0a4c8844c96a430fbd3d7993e39 --unlock 0x2bAaA41d155ad8a4126184950B31F50A1513cE25 --unlock 0x7a3a1c2de64f20eb5e916f40d11b01c441b2a8dc --port 9545
-// Then execute the script as: yarn truffle exec ./scripts/collateral-umip/1_Propose.js --network mainnet-fork --collateral 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2 --fee 0.1 from core
+// Then execute the script as: yarn truffle exec ./scripts/collateral-umip/1_Propose.js --network mainnet-fork --collateral 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2 --fee 0.1 --collateral 0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48 --fee 400 from core
 
 const AddressWhitelist = artifacts.require("AddressWhitelist");
 const Store = artifacts.require("Store");
 const Governor = artifacts.require("Governor");
+const ERC20 = artifacts.require("ERC20");
 
-const argv = require("minimist")(process.argv.slice(), { string: ["collateral", "fee"] });
+const { parseUnits } = require("@ethersproject/units");
+
+const _ = require("lodash");
+
+const argv = require("minimist")(process.argv.slice(), { string: ["collateral", "fee", "decimals"] });
 
 const proposerWallet = "0x2bAaA41d155ad8a4126184950B31F50A1513cE25";
 
+async function getDecimals(collateralAddress, decimalsArg) {
+  const collateral = await ERC20.at(collateralAddress);
+  if (decimalsArg) {
+    console.log(`Using user input decimals: ${decimalsArg} for collateral ${collateralAddress}`);
+    return decimalsArg;
+  } else {
+    try {
+      const decimals = (await collateral.decimals()).toString();
+      console.log(`Using decimals returned by contract: ${decimals} for collateral ${collateralAddress}`);
+      return decimals;
+    } catch (error) {
+      throw "Must provide --decimals if token has no decimals function.";
```

Comment:
i like the intention of this function, and this is really just nit picking but this will end up obscuring the error for everything but the contract having no decimals function. May want to pass through the actual error message by appending error.message. like throw "Try providing --decimals to prevent the following error: " + error.message. 

Also id say wrap that in a `new Error()`

Reply:
+1. Done.